### DISCUSSION TITLE
[Help] Using `zk-nvim` commands from a lua script

### DISCUSSION THREAD

**Discussion by lyndhurst** [zk-org/zk-nvim#147] [Help]
I am trying to build a few functions to suit my needs, but I am having trouble accessing `zk-nvim` commands.

When trying to use `zkNew` in a `lua` script, it does not return a dictionary as indicated in the [linked documentation](https://github.com/mickael-menu/zk/blob/main/docs/editors-integration.md#zknew).

I tried all the following:

```lua
require("zk.api").new()
require("zk").new()
require("zk.commands").get("ZkNew")()
```

They all return a `nil` value, and do nothing when the `dryRun = true` option is passed.

Is it related to #142 ? Does it mean that none of the commands provided by the plugin can be used as described in the [linked documentation](https://github.com/mickael-menu/zk/blob/main/docs/editors-integration.md#zknew) ? Or am I missing something ?

Thanks in advance for your help.

**Comment by tjex** [comment]
If this is still an issue, could you post in here your attempt?

As an example, this is a custom command within lua (using `ZkEdit` instead of `ZkNew`, but the procedure is the same):

```lua
local cmd = require("zk.commands")
local zk = require("zk")

cmd.add("ZkOrphans", function(options)
	options = vim.tbl_extend("force", { orphan = true }, options or {})
	zk.edit(options, { title = "Zk Orphans" })
end)
```
